content/articles/working-with-github's-web-api-in-python/index.md

```diff
@@ -0,0 +1,241 @@
+### Introduction
+This tutorial will teach you how to create a self-contained application that generates a visualization based on the information it obtains. Instead of requesting complete pages, a web application programming interface will be used by your program (API) to get certain information from the web. We'll use data from [GitHub](https://github.com/), a website where programmers can contribute to projects. We'll request information concerning Python projects on GitHub using Github's API.
+
+### Prerequisites
+You'll need some Python skills to continue with this tutorial.
+
+### What you will learn
+- Using an API call to request data
+- Requests installation
+- Keeping track of an API response
+- Using the response dictionary
+- Summing up the top repositories
+
+### Requesting data using an API call 
+GitHub's web API allows you to make API requests for a range of data. Type the following into your web browser URL bar and press **enter** to see how an API call appears like:
+
+```bash
+https://api.github.com/search/repositories?q=language:python&sort=stars
+```
+
+Let’s examine the parts of the call:
+ 1. `https://api.github.com/` - sends the request to the section of GitHub's website that handles API calls.
+ 2. `search/repositories` - informs the API to search across all of GitHub's repositories.
+ 3. `?` - indicates that an argument is about to be passed.
+ 4. `q=`- the character q means for "query", and then we can start working with API by specifying a query with the equal sign. 
+ 5. `language:python` - we specify that we only need data on repositories that use Python as their main language. 
+ 6. `&sort=stars` - the projects are ordered by the number of stars they have gotten.
+
+The response's first several lines are shown in the below snippet:
+
+```json
+{
+  "total_count": 7668509,
+  "incomplete_results": false,
+  "items": [
+    {
+      "id": 54346799,
+      "node_id": "MDEwOlJlcG9zaXRvcnk1NDM0Njc5OQ==",
+      "name": "public-apis",
+      "full_name": "public-apis/public-apis",
+      --snip--
```

Comment:
Not sure if this JSON is complete. Could you please check on this?

Reply:
It in not complete .It is very long hence I decided to cut it short